### DISCUSSION TITLE
Restrict scrolling of canvas

### DISCUSSION THREAD

**Discussion by jerinjohn1412** [xyflow/xyflow#2142] [Q&A]
In our case we create flows that have multiple combinations of branches (nested nodes) and groups, we like the canvas to flow only till the last nodes,  that is only till it vertically scrolls up to the last visible node, and horizontally in a similar manner.

**Comment by moklick** [comment]
You can set the `translateExtent` prop dynamically based on the dimensions of your graph https://reactflow.dev/docs/api/react-flow-props/#flow-view
